sources/packages/backend/apps/api/src/route-controllers/education-program-offering/_tests_/e2e/education-program-offering.students.controller.getOfferingSummaryDetails.e2e-spec.ts

```diff
@@ -0,0 +1,183 @@
+    // Arrange
+    const savedUser = await db.user.save(createFakeUser());
+    const fakeOffering = createFakeEducationProgramOffering({
+      auditUser: savedUser,
+    });
+    const offering = await db.educationProgramOffering.save(fakeOffering);
+    const fakeApplication = createFakeApplication();
+    const application = await applicationRepo.save(fakeApplication);
+    const fakeAssessment = createFakeStudentAssessment({
+      auditUser: savedUser,
+      application,
+      offering,
+    });
+    fakeApplication.currentAssessment = fakeAssessment;
+    await applicationRepo.save(fakeApplication);
+    await assessmentRepo.save(fakeAssessment);
+    await saveFakeApplicationOfferingRequestChange(db, {
+      application,
+    });
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const endpoint = `/students/education-program-offering/offering/${fakeOffering.id}/summary-details?purpose=application-offering-change`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: HttpStatus.NOT_FOUND,
+        message: "Not able to find the Education Program offering.",
+        error: "Not Found",
+      });
+  });
+
+  it("Should throw the Not Found (404) exception when the offering summary purpose is not provided.", async () => {
+    // Arrange
+    const savedUser = await db.user.save(createFakeUser());
+    const fakeOffering = createFakeEducationProgramOffering({
+      auditUser: savedUser,
+    });
+    const offering = await db.educationProgramOffering.save(fakeOffering);
+    const fakeApplication = createFakeApplication({ student });
+    const application = await applicationRepo.save(fakeApplication);
+    const fakeAssessment = createFakeStudentAssessment({
+      auditUser: savedUser,
+      application,
+      offering,
+    });
+    fakeApplication.currentAssessment = fakeAssessment;
+    await applicationRepo.save(fakeApplication);
+    await assessmentRepo.save(fakeAssessment);
+    await saveFakeApplicationOfferingRequestChange(db, {
+      application,
+    });
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    const endpoint = `/students/education-program-offering/offering/${fakeOffering.id}/summary-details`;
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.NOT_FOUND)
+      .expect({
+        statusCode: HttpStatus.NOT_FOUND,
+        message: "Not able to find the Education Program offering.",
+        error: "Not Found",
+      });
+  });
+
+  afterAll(async () => {
+    await app?.close();
+  });
+
+  const deliveryMethod = (offering: EducationProgramOffering) => {
```

Comment:
can be simplified:
![image](https://github.com/bcgov/SIMS/assets/78114138/43829617-e77e-404b-b18c-3674348d246f)